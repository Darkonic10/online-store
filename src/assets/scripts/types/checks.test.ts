import { checkBookId, createElementByTag, formatterUSD, getBookID, getHash, getLocalStorage, getMainAddress, getMainOptions, resetMainOptions } from "./checks"

const fakeLocalStorage = {
  getItem: (key: string) => {
    return 'QuillTree↕SimonSchuster'
  },
  length: 1,

  clear: () => {
    return undefined;
  },

  key: () => {
    return null
  },

  removeItem: () => {
    return null
  },

  setItem: () => {
    return null
  },
}

describe('Check function tests', () => {
  it('should check is this book id exist', () => {
    expect(checkBookId(5)).toBeDefined();
  }, 3000);

  it('should get book id if it exists', () => {
    const options = new Map([['id', '18']]);
    const result = '18'
    expect(getBookID(options)).toMatchSnapshot(result);
  }, 3000);

  it('should return formatted price', () => {
    const price = 17;
    const result = '$17.00'
    expect(formatterUSD.format(price)).toEqual(result);
  }, 3000);

  it('should return clean page address without query parameters', () => {
    const url = '#main-page?maxStock=11&publisher=QuillTree↕SimonSchuster';
    const result = 'main-page'
    expect(getHash(url)).toEqual(result);
    expect(typeof getHash(url) === 'string').toBeTruthy();
  }, 3000);

  it('should return HTML element of given type', () => {
    expect(getLocalStorage(fakeLocalStorage, 'publisher')).toEqual('QuillTree↕SimonSchuster');
  }, 3000);
});

test('throw an error if no such a book was found', () => {
  expect(() => {checkBookId(0)}).toThrowError();
});