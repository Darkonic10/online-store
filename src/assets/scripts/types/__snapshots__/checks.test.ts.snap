// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check function tests should get book id if it exists: 18 1`] = `18`;
